### DISCUSSION TITLE
zig 的包管理器还会再优化吗？

### DISCUSSION THREAD

**Discussion by deatil** [zigcc/forum#131] [General]
zig 的包管理器还会再优化吗？现在的包管理器说起来就没有go的好用，还会更新到更简单些，不要那个什么hash吗？hash只要zig知道就行，虽然是有了包管理器，但是难用啊

**Comment by zhang-chuanfeng** [comment]
hash zig可以自动生成 更新到build.zig.zon。比如这个https://github.com/allyourcodebase/zlib

**Reply by deatil** [reply]
手动填的能自动更新吗

**Comment by jiacai2050** [comment]
`zig fetch --save` 就可以自动生成 hash

**Reply by deatil** [reply]
手动填的好像不行吧

**Comment by jiacai2050** [comment]
像 Rust 的 cargo 是有一个中心的话 crates io 来记录一个包的 checksum，zig 没有这个东西，所有这个 hash 应该是去不掉的

**Reply by deatil** [reply]
像go的不行吗？go的都是下载后计算的

**Reply by jiacai2050** [reply]
go 也有个 sumdb，你本地算完也要和他检验。

https://sum.golang.org/

**Comment by hfghgakjd** [comment]
zig fetch 不是已经不需要填zon文件了吗

**Reply by deatil** [reply]
手动添加的时候
